### PR TITLE
Refactor for TLS handlers

### PR DESCRIPTION
**Pull request by javuto**
## Overview

* Refactor for HTTP handlers, that will allow easier testing.

### CODE REVIEW

File: admin/main.go [hunk 1]
```diff
@@ -416,6 +414,9 @@ func main() {
 		}
 	}()
 
+	// multiple listeners channel
+	finish := make(chan bool)
```
Comment (pmoody-) [inline admin/main.go]: where is this used?

File: logging/dispatch.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	// FIXME allow multiple types of logging
+	if debug {
+		log.Printf("dispatching logs to %s", l.Logging)
+	}
+	l.Log(
+		logType,
+		data,
+		environment,
+		uuid,
+		debug)
```
Comment (pmoody-) [inline logging/dispatch.go]: nit, funny line splitting.

File: logging/dispatch.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	// Refresh last logging request
+	if logType == types.StatusLog {
+		err := l.Nodes.RefreshLastStatus(uuid)
+		if err != nil {
```
Comment (pmoody-) [inline logging/dispatch.go]:
```
if err := l.Nodes.RefreshLastStatus(uuid); err != nil {
}
```

File: logging/dispatch.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+		log.Printf("dispatching queries to %s", l.Logging)
+	}
+	l.QueryLog(
+		types.QueryLog,
+		data,
+		node.Environment,
+		node.UUID,
+		queryData.Name,
+		queryData.Status,
+		debug)
```
Comment (pmoody-) [inline logging/dispatch.go]: same odd line-splitting comment.

File: logging/process.go [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+	"github.com/jmpsec/osctrl/nodes"
+	"github.com/jmpsec/osctrl/types"
+)
+
+// ProcessLogs - Helper to process logs
+func (l *LoggerTLS) ProcessLogs(data json.RawMessage, logType, environment, ipaddress string, debug bool) {
+	// Parse log to extract metadata
+	var logs []types.LogGenericData
+	err := json.Unmarshal(data, &logs)
+	if err != nil {
```
Comment (pmoody-) [inline logging/process.go]:
```
if err := ...; err != nil {
}
```

File: tls/handlers/utils.go [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+	}
+}
+
+// Helper to remove duplicates from array of strings
+func uniq(duplicated []string) []string {
+	keys := make(map[string]bool)
+	result := []string{}
+	for _, entry := range duplicated {
+		if _, value := keys[entry]; !value {
+			keys[entry] = true
```
Comment (pmoody-) [inline tls/handlers/utils.go]: i don't think you actually need this. you don't use it anywhere.

File: admin/main.go (hunk 1 above)
Comment (javuto) [reply admin/main.go]: Hmmm I had this from a long time ago when I was running all services in the same binary, but now I am not using it anymore. Remove!

**Pull request merged**